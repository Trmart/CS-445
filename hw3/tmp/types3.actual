.   .   .   Child:   Id: arwen [line: 26]
.   .   .   Child:   Id: fred [line: 13]
.   Child:   Parm: a of type int [line: 5]
.   Child:   Parm: a of type int [line: 9]
.   Child:   Parm: eowyn of type int [line: 24]
.   Child:   Parm: linda of type int [line: 18]
.   .   Child:   Var: a of type int [line: 11]
.   .   Child:   Var: linda of type int [line: 20]
ERROR(11): Symbol 'a' is already declared at line 9.
ERROR(12): 'a' is a simple variable and cannot be called.
ERROR(13): Cannot use function 'fred' as a variable.
ERROR(16): Symbol 'fred' is already declared at line 9.
ERROR(18): Symbol '(' is already declared at line 5.
ERROR(20): Symbol 'linda' is already declared at line 18.
ERROR(24): Symbol '(' is already declared at line 5.
ERROR(26): Symbol 'arwen' is not declared.
ERROR(9): Symbol 'a' is already declared at line 9.
ERROR(LINKER): A function named 'main()' must be defined.
.   .   hild:   Call: a [line: 12]
.   hild:   Compound [line: 10]
.   hild:   Compound [line: 19]
.   hild:   Compound [line: 25]
.   hild:   Compound [line: 6]
.   .   .   hild:   Const 4 [line: 13]
.   .   .   hild:   Const 4 [line: 26]
.   .   hild:   Op: + [line: 26]
Number of errors: 10
Number of warnings: 3
Sibling: 1  Func: ( returns type void [line: 5]
.   .   Sibling: 1  Op: + [line: 13]
.   Sibling: 1  Parm: a of type int [line: 9]
Sibling: 2  Func: fred returns type int [line: 9]
Sibling: 3  Var: fred of type int [line: 16]
Sibling: 4  Func: ( returns type void [line: 18]
Sibling: 5  Var: galadriel of type int [line: 23]
Sibling: 6  Func: ( returns type void [line: 24]
Var: lund of type int [line: 3]
WARNING(18): The variable 'linda' seems not to be used.
WARNING(24): The variable 'eowyn' seems not to be used.
WARNING(5): The variable 'a' seems not to be used.
