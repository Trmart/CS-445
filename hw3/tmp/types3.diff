1,14c1,34
< ERROR(11): Symbol 'a' is already declared at line 9.
< ERROR(12): 'a' is a simple variable and cannot be called.
< ERROR(13): Cannot use function 'fred' as a variable.
< ERROR(16): Symbol 'fred' is already declared at line 9.
< ERROR(20): Symbol 'linda' is already declared at line 18.
< ERROR(26): Cannot use function 'arwen' as a variable.
< ERROR(5): Symbol 'lund' is already declared at line 3.
< ERROR(9): Symbol 'a' is already declared at line 9.
< ERROR(LINKER): A function named 'main()' must be defined.
< Number of errors: 9
< Number of warnings: 3
< WARNING(18): The variable 'linda' seems not to be used.
< WARNING(24): The variable 'eowyn' seems not to be used.
< WARNING(5): The variable 'a' seems not to be used.
---
> .   .   .   Child: 0  Id: arwen of type undefined [line: 26]
> .   .   .   Child: 0  Id: fred of type undefined [line: 13]
> .   Child: 0  Parm: a of type int [line: 5]
> .   Child: 0  Parm: a of type int [line: 9]
> .   Child: 0  Parm: eowyn of type int [line: 24]
> .   Child: 0  Parm: linda of type int [line: 18]
> .   .   Child: 0  Var: a of type int [line: 11]
> .   .   Child: 0  Var: linda of type int [line: 20]
> .   .   Child: 1  Call: a [line: 12]
> .   Child: 1  Compound [line: 10]
> .   Child: 1  Compound [line: 19]
> .   Child: 1  Compound [line: 25]
> .   Child: 1  Compound [line: 6]
> .   .   .   Child: 1  Const 4 of type int [line: 13]
> .   .   .   Child: 1  Const 4 of type int [line: 26]
> .   .   Child: 1  Op: + of type int [line: 26]
> ERROR(13): Symbol 'fred' is not declared.
> ERROR(26): Symbol 'arwen' is not declared.
> Number of errors: 2
> Number of warnings: 5
> Sibling: 1  Func: lund returns type void [line: 5]
> .   .   Sibling: 1  Op: + of type int [line: 13]
> .   Sibling: 1  Parm: a of type int [line: 9]
> Sibling: 2  Func: fred returns type int [line: 9]
> Sibling: 3  Var: fred of type int [line: 16]
> Sibling: 4  Func: linda returns type void [line: 18]
> Sibling: 5  Var: galadriel of type int [line: 23]
> Sibling: 6  Func: arwen returns type void [line: 24]
> Var: lund of type int [line: 3]
> WARNING(11): The variable 'a' seems not to be used.
> WARNING(16): The variable 'fred' seems not to be used.
> WARNING(20): The variable 'linda' seems not to be used.
> WARNING(23): The variable 'galadriel' seems not to be used.
> WARNING(3): The variable 'lund' seems not to be used.
