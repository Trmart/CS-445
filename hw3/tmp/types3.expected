ERROR(11): Symbol 'a' is already declared at line 9.
ERROR(12): 'a' is a simple variable and cannot be called.
ERROR(13): Cannot use function 'fred' as a variable.
ERROR(16): Symbol 'fred' is already declared at line 9.
ERROR(20): Symbol 'linda' is already declared at line 18.
ERROR(26): Cannot use function 'arwen' as a variable.
ERROR(5): Symbol 'lund' is already declared at line 3.
ERROR(9): Symbol 'a' is already declared at line 9.
ERROR(LINKER): A function named 'main()' must be defined.
Number of errors: 9
Number of warnings: 3
WARNING(18): The variable 'linda' seems not to be used.
WARNING(24): The variable 'eowyn' seems not to be used.
WARNING(5): The variable 'a' seems not to be used.
